framework/packages/abstract-core/src/native/module_factory.rs

```diff
@@ -57,18 +59,25 @@ pub enum ExecuteMsg {
         ans_host_address: Option<String>,
         version_control_address: Option<String>,
     },
-    /// Installs a module on the Account
-    InstallModule {
-        // Module details
-        module: ModuleInfo,
-        init_msg: Option<Binary>,
+    InstallModules {
+        // Module info and init message
+        modules: Vec<(ModuleInfo, Option<Binary>)>,
+        // TODO: alternative API?
+        // modules: Vec<InstallModule>,
     },
     UpdateFactoryBinaryMsgs {
         to_add: Vec<(ModuleInfo, Binary)>,
         to_remove: Vec<ModuleInfo>,
     },
 }
 
+// TODO: do we want to move to that to have more user-friendly api?
+// #[cosmwasm_schema::cw_serde]
+// pub struct InstallModule {
+//     pub module: ModuleInfo,
+//     pub init_msg: Option<Binary>
+// }
+
```

Comment:
@CyberHoward what do you think? Or array of tuples is good enough?

Reply:
I'd like this yes, then we can easily add a `coins: Vec<Coin>` in there as well for funding the module on instantiation.

You could also make it non-exhaustive